src/test/java/searchengine/booleanretrieval/DataSelectionTest.java

```diff
@@ -0,0 +1,81 @@
+package searchengine.booleanretrieval;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.mockito.configuration.IMockitoConfiguration;
+import searchengine.ISqlConnection;
+import searchengine.vsm.TFIDFDocument;
+
+import javax.xml.crypto.Data;
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class DataSelectionTest {
+
+    /** Handles the Arrange part of the tests (can't use @BeforeAll since it will give race condition errors) */
+    static DataSelection setup() {
+        ISqlConnection connection = mock(ISqlConnection.class);
+        Statement statement = mock(Statement.class);
+        ResultSet resultSet = mock(ResultSet.class);
+        try {
+            when(connection.createStatement()).thenReturn(statement);
+            when(statement.executeQuery(anyString())).thenReturn(resultSet);
+            when(resultSet.next()).thenReturn(true, true, true, false);
+            when(resultSet.getString("wordname")).thenReturn("lorem", "ipsum", "dolor");
+            when(resultSet.getInt("amount")).thenReturn(5, 4, 3);
+            when(resultSet.getString("articletitle")).thenReturn("testarticle1", "testarticle1","testarticle2");
+            when(resultSet.getString("filepath")).thenReturn("/test/testarticle1", "testarticle1","/test/testarticle2");
+
+            return new DataSelection(connection);
+        }
+        catch (Exception exception){
+            fail(exception.getMessage());
+        }
+        return null;
+    }
+
+    @Test
+    void retrieveDocuments_mockedArticles_articleCount() {
```

Comment:
```suggestion
    @Test
    This tests uses the retrieveDocuments function. It should return two articles because there are three articles, but two titles are identicals - the reason being that DISTINCT is used. It's mocked here: when(resultSet.getString("articletitle")).thenReturn("testarticle1", "testarticle1","testarticle2");
    void retrieveDocuments_mockedArticles_articleCount() {
```

Reply:
Det er ikke sådan det bliver mocked og derfor man kun får to. Tror du overkomplicere det lidt, men skal nok tilføje en comment i setup så man får forklaret hvad der bliver mocked.